pkg/specgen/generate/container.go

```diff
@@ -117,6 +117,9 @@ func CompleteSpec(ctx context.Context, r *libpod.Runtime, s *specgen.SpecGenerat
 	if err != nil {
 		return nil, fmt.Errorf("parsing fields in containers.conf: %w", err)
 	}
+	if !s.Terminal {
+		delete(defaultEnvs, "TERM")
```

Comment:
Why do we need to delete this at all here?
I think if TERM is set in containers.conf by a user we should not remove it.

Basically I think it would be best to not add TERM at all in the defaults in c/common.
So I think make a PR in c/common to remove TERM from the defaults. And then here we do something like
```
if s.Terminal && TERM env does not exists {
		add "TERM=xterm"
```